serviceRegistry_test.go

```diff
@@ -57,3 +75,26 @@ func TestRedefineCategoryList(t *testing.T) {
 	_, categoryPresent = actual["foo"]
 	assert.True(t, categoryPresent, "foo category should be present")
 }
+
+func TestWatchCategories(t *testing.T) {
+	initLogs(os.Stdout, os.Stdout, os.Stderr)
+
+	fooCategory := client.Node{Key: "/ft/healthcheck-categories/foo", Dir: true, Nodes: client.Nodes{}}
+	categoryNodes := client.Nodes{&fooCategory}
+	categoryFolder := client.Node{Key: "/ft/healthcheck-categories", Dir: true, Nodes: categoryNodes}
+
+	etcd := TestEtcdKeysAPI{&client.Response{Node: &categoryFolder}, &TestWatcher{&client.Response{Node: &categoryFolder}}}
+
+	registry := NewCocoServiceRegistry(etcd, "127.0.0.1", nil)
+	registry.etcdInterval = time.Second
+
+	go registry.watchCategories()
+	time.Sleep(time.Duration(2) * time.Second)
+
+	actual := registry.categories()
+	assert.Len(t, actual, 2, "category list")
```

Comment:
Should be `assert.Len(t, 2, actual, ...)` so that it's less confusing when it breaks